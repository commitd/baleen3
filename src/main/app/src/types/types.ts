/*-
 * #%L
 * Baleen 3
 * %%
 * Copyright (C) 2020 Dstl
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
import { Annot8ComponentDescriptor } from './base-types'
import {
  Annot8ComponentInfo,
  Measurement,
  PipelineTemplate,
  SettingsSchema,
  WithDescription,
  WithName,
  ErrorConfiguration,
} from './server-types'

/**
 * Index of metrics measurements
 */
export interface MeasurementContainer {
  [key: string]: Measurement[]
}

/**
 * Holds all the classes that have metrics associated with them, and their associated measurements
 */
export interface MetricsContainer {
  [key: string]: MeasurementContainer
}

export interface SettingsSchemaExtended extends SettingsSchema {
  /**
   * String version of json uiSchema
   *
   */
  uiSchema: string
}

export interface ComponentInfo extends Annot8ComponentInfo {
  name: string
}

/**
 * Map of server components
 */
export interface ComponentMap {
  [key: string]: ComponentInfo
}

/**
 * Map of settings schema
 */
export interface SettingsMap {
  [key: string]: SettingsSchema
}

/**
 * The full set of customisation details for the server.
 */
export interface ServerDetails {
  /**
   * Index by class name of ComponentInfo that represent the processors available on the server that can be used in pipelines.
   */
  processors: ComponentMap
  /**
   * Index by class name of ComponentInfo that represent the sources available on the server that can be used in pipelines.
   */
  sources: ComponentMap
  /**
   * Index by class name of SettingsSchema that are the settings classes of all sources and processors available on the server that can be used in pipelines.
   */
  settings: SettingsMap
  /**
   * Index by class name of ComponentInfo that represent the orderers available on the server that can be used to order pipelines.
   * <p>
   * NB this is currently generated from a strings to simulate a ComponentInfos
   */
  orderers: ComponentMap
  /**
   * Pipeline templates that can be used to start the process of creating a pipeline
   */
  templates: PipelineTemplate[]
}

export interface ComponentMetadata {
  id: string
  descriptor: Annot8ComponentDescriptor
  info: ComponentInfo | undefined
  schema: SettingsSchema | undefined
  valid: boolean
  configured: boolean
}

/**
 * Mapping of Pipeline descriptor to include extra metadata
 *
 */
export interface PipelineHeaderDescriptor extends WithName, WithDescription {}

/**
 * Mapping of Pipeline descriptor to include extra metadata
 *
 */
export interface PipelineViewDescriptor extends PipelineHeaderDescriptor {
  sourceOrder: string[]
  processorOrder: string[]
  components: Record<string, ComponentMetadata>
  errorConfiguration: ErrorConfiguration
}

/**
 * Utility tracking interface to help maintain the UI key state
 */
export interface PipelineEditDescriptor extends PipelineViewDescriptor {
  nameValid: boolean
  orderer: string
  errors: string[]
}

export type ComponentType = 'Processor' | 'Source'

export const NO_SETTINGS: 'io.annot8.api.settings.NoSettings' =
  'io.annot8.api.settings.NoSettings'

export const NO_OP_ORDERER: 'io.annot8.api.pipelines.NoOpOrderer' =
  'io.annot8.api.pipelines.NoOpOrderer'
