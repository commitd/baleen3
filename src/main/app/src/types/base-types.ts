/*-
 * #%L
 * Baleen 3
 * %%
 * Copyright (C) 2020 Dstl
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
// We remove the generics for the Settings type from these as we never know what they are
// so do not help with any compile time errors

export interface ComponentDescriptor {
  name: string
  settings?: object
}

export interface Annot8ComponentDescriptor {
  [_class: string]: ComponentDescriptor
}

// Types required to complete generated server types
// eslint-disable-next-line @typescript-eslint/no-empty-interface, @typescript-eslint/no-unused-vars-experimental
export interface SourceDescriptor extends Annot8ComponentDescriptor {}
// eslint-disable-next-line @typescript-eslint/no-empty-interface, @typescript-eslint/no-unused-vars-experimental
export interface ProcessorDescriptor extends Annot8ComponentDescriptor {}

/**
 * Interface for describing known metrics, to give better display.
 */
export interface MetricMetadata {
  name?: string
  description?: string
  units?: string
  transform?: (v: number) => number
}

/**
 * Provided stacktrace type to correct optional properties
 */
export interface StackTrace {
  classLoaderName?: string
  className: string
  fileName?: string
  lineNumber: number
  methodName: string
  moduleName?: string
  moduleVersion?: string
  nativeMethod?: boolean
}

/**
 * Provided throwable type to correct optional properties
 */
export interface Throwable {
  cause?: Throwable
  localizedMessage: string
  message: string
  stackTrace: StackTrace[]
  suppressed?: Throwable[]
}
