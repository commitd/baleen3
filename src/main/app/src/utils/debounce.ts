/*-
 * #%L
 * Baleen 3
 * %%
 * Copyright (C) 2020 Dstl
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
/* eslint-disable @typescript-eslint/no-explicit-any */
export type Method = (...args: any[]) => void

export function debounce<F extends Method>(
  func: F,
  wait = 200
): (this: ThisParameterType<F>, ...args: Parameters<F>) => void {
  let timeoutId: ReturnType<typeof setTimeout> | undefined

  return function (this: ThisParameterType<F>, ...args: Parameters<F>): void {
    // eslint-disable-next-line @typescript-eslint/no-this-alias
    const context = this

    if (timeoutId !== undefined) {
      clearTimeout(timeoutId)
    }
    timeoutId = setTimeout(() => {
      timeoutId = undefined
      func.apply(context, args)
    }, wait)
  }
}
