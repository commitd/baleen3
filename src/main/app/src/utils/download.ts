/*-
 * #%L
 * Baleen 3
 * %%
 * Copyright (C) 2020 Dstl
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const downloadJson = (name: string, data: any): void => {
  download(
    name + '.json',
    'data:text/json;charset=utf-8,',
    JSON.stringify(data, null, 2)
  )
}

export const download = (
  name: string,
  mediaType: string,
  data: string
): void => {
  const dataStr = mediaType + encodeURIComponent(data)
  const downloadAnchorNode = document.createElement('a')
  downloadAnchorNode.setAttribute('href', dataStr)
  downloadAnchorNode.setAttribute('download', name)
  document.body.appendChild(downloadAnchorNode) // required for firefox
  downloadAnchorNode.click()
  downloadAnchorNode.remove()
}
