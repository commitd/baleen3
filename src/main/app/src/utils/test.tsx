/*-
 * #%L
 * Baleen 3
 * %%
 * Copyright (C) 2020 Dstl
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
import { StylesProvider } from '@material-ui/styles'
import {
  createHistory,
  createMemorySource,
  History,
  LocationProvider,
} from '@reach/router'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { GenerateId } from 'jss'
import React from 'react'
import { ThemeProvider } from '../components/ThemeProvider/ThemeProvider'

const generateClassName: GenerateId = (rule, styleSheet) => {
  const prefix =
    styleSheet === undefined || styleSheet.options.classNamePrefix === undefined
      ? ''
      : styleSheet.options.classNamePrefix
  return `${prefix}-${rule.key}`
}

const LightTheme: React.FC<{ readonly children?: React.ReactNode }> = ({
  children,
}) => (
  <StylesProvider generateClassName={generateClassName}>
    <ThemeProvider choice="light">{children}</ThemeProvider>
  </StylesProvider>
)

const DarkTheme: React.FC<{ readonly children?: React.ReactNode }> = ({
  children,
}) => (
  <StylesProvider generateClassName={generateClassName}>
    <ThemeProvider choice="dark">{children}</ThemeProvider>
  </StylesProvider>
)

export const renderPlain = render

export const renderLight = (
  ui: Readonly<React.ReactElement>,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult => render(ui, { wrapper: LightTheme, ...options })

export const renderDark = (
  ui: Readonly<React.ReactElement>,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult => render(ui, { wrapper: DarkTheme, ...options })

export function renderWithRouter(
  ui: Readonly<React.ReactElement>,
  route = '/',
  options?: Omit<RenderOptions, 'queries'>,
  delegate = renderLight
): RenderResult & { history: History } {
  const history = createHistory(createMemorySource(route))
  return {
    ...delegate(
      <LocationProvider history={history}>{ui}</LocationProvider>,
      options
    ),
    history,
  }
}
// re-export everything
export * from '@testing-library/react'
// override render method
export { renderLight as render }
export { userEvent }
