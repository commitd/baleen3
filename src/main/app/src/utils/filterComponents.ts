/*-
 * #%L
 * Baleen 3
 * %%
 * Copyright (C) 2020 Dstl
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
import { ComponentInfo, ComponentMap } from '../types'
import { parseSearch, standardize } from './text'

export const filterComponents = (
  map: ComponentMap,
  search: string
): Array<ComponentInfo> => {
  const values = Object.values(map)
  if (search.trim() === '') {
    return values
  }

  const searchTerms = parseSearch(search)

  return values.filter(
    (value) =>
      filterBy('artifact', value, searchTerms) ||
      filterBy('description', value, searchTerms) ||
      filterByName(value, searchTerms) ||
      filterByTags(value, searchTerms)
  )
}

function filterBy(
  key: keyof Omit<ComponentInfo, 'tags'>,
  value: ComponentInfo,
  terms: string[]
): boolean {
  // false positive security/detect-object-injection
  // eslint-disable-next-line
  const property = value[key]
  if (property === undefined) {
    return false
  }
  return terms.some((term) => standardize(property).includes(term))
}

function filterByName(value: ComponentInfo, terms: string[]): boolean {
  const property = standardize(value.name)
  return terms.some((term) => property.includes(term))
}

function filterByTags(value: ComponentInfo, terms: string[]): boolean {
  const tags = value.tags?.map((tag) => tag.toLowerCase())
  return terms.some((term) => tags?.includes(term))
}
