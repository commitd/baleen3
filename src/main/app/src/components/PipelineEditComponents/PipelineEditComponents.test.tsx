/*-
 * #%L
 * Baleen 3
 * %%
 * Copyright (C) 2020 Dstl
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
import React from 'react'
import { render, renderDark } from '../../utils/test'
import { PipelineEditComponents } from '.'
import { exampleProcessorMetadata } from '../../types/examples'

it('renders without error', () => {
  const { asFragment } = render(
    <PipelineEditComponents
      id="test"
      type="Processor"
      components={[exampleProcessorMetadata]}
      addComponents={jest.fn()}
      removeComponent={jest.fn()}
      moveComponent={jest.fn()}
      setComponentName={jest.fn()}
      setComponentSettings={jest.fn()}
    />
  )
  expect(asFragment()).toMatchSnapshot()
})

it('renders dark without error', () => {
  const { asFragment } = renderDark(
    <PipelineEditComponents
      id="test"
      type="Processor"
      components={[exampleProcessorMetadata]}
      addComponents={jest.fn()}
      removeComponent={jest.fn()}
      moveComponent={jest.fn()}
      setComponentName={jest.fn()}
      setComponentSettings={jest.fn()}
    />
  )
  expect(asFragment()).toMatchSnapshot()
})
