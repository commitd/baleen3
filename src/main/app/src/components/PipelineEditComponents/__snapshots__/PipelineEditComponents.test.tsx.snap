// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dark without error 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
      data-rbd-droppable-context-id="1"
      data-rbd-droppable-id="test"
    >
      <div
        class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
        data-rbd-draggable-context-id="1"
        data-rbd-draggable-id="modjewrijfjwo"
      >
        <button
          aria-label="Insert Processors"
          class="MuiButtonBase-root"
          tabindex="0"
          title="Insert Processors"
          type="button"
        >
          <div
            class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
          >
            <div
              class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root Styled(Styled(MuiBox))-root Styled(Styled(MuiBox))-root"
            />
          </div>
          <div
            class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
            style="opacity: 0; visibility: hidden;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          aria-describedby="rbd-hidden-text-1-hidden-text-3"
          class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
          data-rbd-drag-handle-context-id="1"
          data-rbd-drag-handle-draggable-id="modjewrijfjwo"
          draggable="false"
          role="button"
          tabindex="0"
        >
          <div
            tabindex="-1"
          >
            <div
              class="MuiPaper-root MuiCard-root WithStyles(ForwardRef(Card))-root WithStyles(ForwardRef(Card))-root MuiPaper-elevation1 MuiPaper-rounded"
              tabindex="0"
            >
              <div
                class="MuiCardHeader-root WithStyles(ForwardRef(CardHeader))-root WithStyles(ForwardRef(CardHeader))-root"
              >
                <div
                  class="MuiCardHeader-content"
                >
                  <span
                    class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                  >
                    Geo
                  </span>
                  <span
                    class="MuiTypography-root MuiCardHeader-subheader MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                  >
                    MGRS
                  </span>
                </div>
                <div
                  class="MuiCardHeader-action"
                >
                  <div
                    style="opacity: 0; visibility: hidden;"
                  >
                    <button
                      aria-label="Edit name"
                      class="MuiButtonBase-root MuiIconButton-root WithNoOpener-root WithNoOpener-root"
                      tabindex="0"
                      title="Edit name"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM21.41 6.34l-3.75-3.75-2.53 2.54 3.75 3.75 2.53-2.54z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                    <button
                      aria-label="Edit settings"
                      class="MuiButtonBase-root MuiIconButton-root WithNoOpener-root WithNoOpener-root"
                      tabindex="0"
                      title="Edit Settings"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19.44 12.99l-.01.02c.04-.33.08-.67.08-1.01 0-.34-.03-.66-.07-.99l.01.02 2.44-1.92-2.43-4.22-2.87 1.16.01.01c-.52-.4-1.09-.74-1.71-1h.01L14.44 2H9.57l-.44 3.07h.01c-.62.26-1.19.6-1.71 1l.01-.01-2.88-1.17-2.44 4.22 2.44 1.92.01-.02c-.04.33-.07.65-.07.99 0 .34.03.68.08 1.01l-.01-.02-2.1 1.65-.33.26 2.43 4.2 2.88-1.15-.02-.04c.53.41 1.1.75 1.73 1.01h-.03L9.58 22h4.85s.03-.18.06-.42l.38-2.65h-.01c.62-.26 1.2-.6 1.73-1.01l-.02.04 2.88 1.15 2.43-4.2s-.14-.12-.33-.26l-2.11-1.66zM12 15.5c-1.93 0-3.5-1.57-3.5-3.5s1.57-3.5 3.5-3.5 3.5 1.57 3.5 3.5-1.57 3.5-3.5 3.5z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                    <button
                      aria-label="Delete"
                      class="MuiButtonBase-root MuiIconButton-root WithNoOpener-root WithNoOpener-root"
                      tabindex="0"
                      title="Delete"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M6 21h12V7H6v14zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="MuiCardContent-root"
              >
                <div
                  class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
                >
                  <div
                    class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
                  >
                    <p
                      class="MuiTypography-root WithStyles(ForwardRef(Typography))-root WithStyles(ForwardRef(Typography))-root WrappedTypography-root WrappedTypography-root MuiTypography-body1"
                    >
                      Extract MGRS coordinates, optionally ignoring MGRS coordinates that could be dates
                    </p>
                  </div>
                  <div
                    class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        aria-label="Insert Processors"
        class="MuiButtonBase-root"
        tabindex="0"
        title="Insert Processors"
        type="button"
      >
        <div
          class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
        >
          <div
            class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root Styled(Styled(MuiBox))-root Styled(Styled(MuiBox))-root"
          />
        </div>
        <div
          class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
          style="opacity: 0; visibility: hidden;"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders without error 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
      data-rbd-droppable-context-id="0"
      data-rbd-droppable-id="test"
    >
      <div
        class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
        data-rbd-draggable-context-id="0"
        data-rbd-draggable-id="modjewrijfjwo"
      >
        <button
          aria-label="Insert Processors"
          class="MuiButtonBase-root"
          tabindex="0"
          title="Insert Processors"
          type="button"
        >
          <div
            class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
          >
            <div
              class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root Styled(Styled(MuiBox))-root Styled(Styled(MuiBox))-root"
            />
          </div>
          <div
            class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
            style="opacity: 0; visibility: hidden;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          aria-describedby="rbd-hidden-text-0-hidden-text-0"
          class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
          data-rbd-drag-handle-context-id="0"
          data-rbd-drag-handle-draggable-id="modjewrijfjwo"
          draggable="false"
          role="button"
          tabindex="0"
        >
          <div
            tabindex="-1"
          >
            <div
              class="MuiPaper-root MuiCard-root WithStyles(ForwardRef(Card))-root WithStyles(ForwardRef(Card))-root MuiPaper-elevation1 MuiPaper-rounded"
              tabindex="0"
            >
              <div
                class="MuiCardHeader-root WithStyles(ForwardRef(CardHeader))-root WithStyles(ForwardRef(CardHeader))-root"
              >
                <div
                  class="MuiCardHeader-content"
                >
                  <span
                    class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                  >
                    Geo
                  </span>
                  <span
                    class="MuiTypography-root MuiCardHeader-subheader MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                  >
                    MGRS
                  </span>
                </div>
                <div
                  class="MuiCardHeader-action"
                >
                  <div
                    style="opacity: 0; visibility: hidden;"
                  >
                    <button
                      aria-label="Edit name"
                      class="MuiButtonBase-root MuiIconButton-root WithNoOpener-root WithNoOpener-root"
                      tabindex="0"
                      title="Edit name"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM21.41 6.34l-3.75-3.75-2.53 2.54 3.75 3.75 2.53-2.54z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                    <button
                      aria-label="Edit settings"
                      class="MuiButtonBase-root MuiIconButton-root WithNoOpener-root WithNoOpener-root"
                      tabindex="0"
                      title="Edit Settings"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19.44 12.99l-.01.02c.04-.33.08-.67.08-1.01 0-.34-.03-.66-.07-.99l.01.02 2.44-1.92-2.43-4.22-2.87 1.16.01.01c-.52-.4-1.09-.74-1.71-1h.01L14.44 2H9.57l-.44 3.07h.01c-.62.26-1.19.6-1.71 1l.01-.01-2.88-1.17-2.44 4.22 2.44 1.92.01-.02c-.04.33-.07.65-.07.99 0 .34.03.68.08 1.01l-.01-.02-2.1 1.65-.33.26 2.43 4.2 2.88-1.15-.02-.04c.53.41 1.1.75 1.73 1.01h-.03L9.58 22h4.85s.03-.18.06-.42l.38-2.65h-.01c.62-.26 1.2-.6 1.73-1.01l-.02.04 2.88 1.15 2.43-4.2s-.14-.12-.33-.26l-2.11-1.66zM12 15.5c-1.93 0-3.5-1.57-3.5-3.5s1.57-3.5 3.5-3.5 3.5 1.57 3.5 3.5-1.57 3.5-3.5 3.5z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                    <button
                      aria-label="Delete"
                      class="MuiButtonBase-root MuiIconButton-root WithNoOpener-root WithNoOpener-root"
                      tabindex="0"
                      title="Delete"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M6 21h12V7H6v14zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="MuiCardContent-root"
              >
                <div
                  class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
                >
                  <div
                    class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
                  >
                    <p
                      class="MuiTypography-root WithStyles(ForwardRef(Typography))-root WithStyles(ForwardRef(Typography))-root WrappedTypography-root WrappedTypography-root MuiTypography-body1"
                    >
                      Extract MGRS coordinates, optionally ignoring MGRS coordinates that could be dates
                    </p>
                  </div>
                  <div
                    class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        aria-label="Insert Processors"
        class="MuiButtonBase-root"
        tabindex="0"
        title="Insert Processors"
        type="button"
      >
        <div
          class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
        >
          <div
            class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root Styled(Styled(MuiBox))-root Styled(Styled(MuiBox))-root"
          />
        </div>
        <div
          class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
          style="opacity: 0; visibility: hidden;"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
